sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/AlertTriggering/MetricAnomalyAlertConfiguration.cs

```diff
@@ -47,12 +47,12 @@ internal MetricAnomalyAlertConfiguration(string detectionConfigurationId, Metric
         /// The identifier of the anomaly detection configuration to which this configuration applies.
         /// </summary>
         [CodeGenMember("AnomalyDetectionConfigurationId")]
-        public string DetectionConfigurationId { get; }
+        public string DetectionConfigurationId { get; set; }
```

Comment:
if my assumption of codegen serializing with custom type is wrong then this is ok. If it is right, shouldn't there be a patch method for this property?

Reply:
I still haven't figured out what you were trying to say, but I think I explained the issue properly in the other PR.

In fact, the service doesn't have a patch version for `MetricAnomalyAlertConfiguration`. This type is not used directly in Create/Update methods. Actually, it is part of a collection (an array in the REST API):

```cs
// Type passed to the Create/Update methods
public class AnomalyAlertConfiguration
{
  public IList<MetricAnomalyAlertConfiguration> AlertConfigurations { get; }

  internal <model-name>Patch GetPatchModel()
  {
    ...
  }
}
```

The service does not support a "partial" array update. According to Johan, this is what's expected according to the JSON "merge-patch" format, used by the MA service. So every time a user wants to update this array, all of its members must be specified in their entirety. That's why they didn't add a patch model for this type.